components/parquet_ext/src/async_arrow_writer.rs

```diff
@@ -0,0 +1,121 @@
+// Copyright 2022 CeresDB Project Authors. Licensed under Apache-2.0.
+
+// Copyright 2023 CeresDB Project Authors. Licensed under Apache-2.0.
+
+//! Async arrow writer for parquet file.
+
+use std::{
+    io::Write,
+    sync::{Arc, Mutex},
+};
+
+use arrow::{datatypes::SchemaRef, record_batch::RecordBatch};
+use parquet::{
+    arrow::ArrowWriter,
+    errors::{ParquetError, Result},
+    file::properties::WriterProperties,
+    format::{FileMetaData, KeyValue},
+};
+use tokio::io::{AsyncWrite, AsyncWriteExt};
+
+#[derive(Clone, Default)]
+pub struct SharedBuffer {
+    /// The buffer to store the data to be written.
+    ///
+    /// The lock is used to obtain internal mutability, so no worry about the
+    /// lock contention.
+    buffer: Arc<Mutex<Vec<u8>>>,
+}
+
+impl Write for SharedBuffer {
+    fn write(&mut self, buf: &[u8]) -> std::io::Result<usize> {
+        let mut buffer = self.buffer.lock().unwrap();
+        Write::write(&mut *buffer, buf)
+    }
+
+    fn flush(&mut self) -> std::io::Result<()> {
+        let mut buffer = self.buffer.lock().unwrap();
+        Write::flush(&mut *buffer)
+    }
+}
+
+/// Async arrow writer for parquet file with a buffer whose size limit is
```

Comment:
I guess you should explain how this async writer is implemented based on a sync one.

And `whose size limit is configurable` seems useless.